### PR TITLE
Generate xunit runner config automatically

### PR DESCRIPTION
**Pull request by tarekgh**
The change here is to enumerate the assemblies in the runtime folder and generate the binding elements in the config file.
we are getting rid of the template we used to generate the config file from

### CODE REVIEW

**Comment by tarekgh** [comment]
@weshaggard this is the change you have requested me to do for generating the config file from the runtime assemblies. no worries if you already became off so I can find someone else to review it. thanks

**Comment by tarekgh** [comment]
@ericstj @joperezr could one of you have a look please as @weshaggard became off? this change is to automate generating the xunit runner config file. Thanks.

**Comment by ericstj** [comment]
Any reason your couldn't use the GenerateBindingRedirect task that @karajas already added to buildtools?

**Comment by tarekgh** [comment]
@ericstj I didn't think about it. so I'll try to look at using GenerateBindingRedirect. thanks for pointing to it.

**Comment by ericstj** [comment]
We discussed how this is different than the GenerateBindingRedirect task because it is putting in the codebase.  I'm ok with this assuming you reconcile the two targets (make sure they're using roughly the same code) then you can replace the one in buildtools with yours when you reconcile.

I think that this whole approach creates a test gap, we should track that in a issue.

**Comment by tarekgh** [comment]
Created issue https://github.com/dotnet/corefx/issues/16235 for the failed CI tests

File: src/Tools/CoreFx.Tools/GenerateBindingRedirect.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        [Required]
+        public ITaskItem[] Assemblies { get; set; }
+
+        [Required]
+        public ITaskItem[] Executables { get; set; }
+
+        [Required]
+        public string OutputPath { get; set; }
+
+        [Required]
```
Comment (weshaggard) [inline src/Tools/CoreFx.Tools/GenerateBindingRedirect.cs]: nit: but we probably should make some of these properties like this one not-required, as making them Required cause older targets files to break if all the other parameters match.

File: src/src.builds [hunk 2]
```diff
@@ -32,4 +33,18 @@
                       DepsExceptions="@(ExceptionForDepsJson)"
                       OutputPath="$(_OutputTestSharedFrameworkDepsPath)"/>
   </Target>
+
+  <Target Name="GenerateXUnitRunnerConfigFile" AfterTargets="BuildAllProjects" Condition="'$(TargetGroup)' == 'netfx'">
```
Comment (weshaggard) [inline src/src.builds]: Is TargetGroup netfx correct here? Should it be _bc_TargetGroup? (perhaps they are always the same for builds files).

Comment (tarekgh) [inline src/src.builds]: I confirmed this is working with TargetGroup

File: src/Tools/CoreFx.Tools/GenerateBindingRedirect.cs (hunk 1 above)
Comment (tarekgh) [inline src/Tools/CoreFx.Tools/GenerateBindingRedirect.cs]: I am not seeing any other targets using this task. we should eliminate the other copy from buildtools

**Pull request merged**